math/src/unsigned_integer/element.rs

```diff
@@ -597,9 +597,22 @@ impl<const NUM_LIMBS: usize> UnsignedInteger<NUM_LIMBS> {
         (UnsignedInteger { limbs }, carry > 0)
     }
 
-    /// Returns the double of `self`.
     pub fn double(a: &UnsignedInteger<NUM_LIMBS>) -> (UnsignedInteger<NUM_LIMBS>, bool) {
-        Self::add(a, a)
+        let mut cloned = (*a).clone();
```

Comment:
This should fix clippy:
```suggestion
        let mut cloned = *a;
```